src/tokenizedSubaccounts/PPTSA.sol

```diff
@@ -0,0 +1,492 @@
+    uint maxFee;
+    IRfqModule.TradeData[] memory makerTrades;
+    if (extraData.length == 0) {
+      IRfqModule.RfqOrder memory makerOrder = abi.decode(action.data, (IRfqModule.RfqOrder));
+      maxFee = makerOrder.maxFee;
+      makerTrades = makerOrder.trades;
+    } else {
+      IRfqModule.TakerOrder memory takerOrder = abi.decode(action.data, (IRfqModule.TakerOrder));
+      if (keccak256(extraData) != takerOrder.orderHash) {
+        revert PPT_TradeDataDoesNotMatchOrderHash();
+      }
+      maxFee = takerOrder.maxFee;
+      makerTrades = abi.decode(extraData, (IRfqModule.TradeData[]));
+    }
+    if (makerTrades[0].asset != address(_getPPTSAStorage().baseCollateralAddresses.optionAsset)) {
+      revert PPT_InvalidAsset();
+    }
+
+    (StrikeData memory lowerStrike, StrikeData memory higherStrike) = _createStrikes(makerTrades);
+    _verifyHigherStrikeAmount(higherStrike.tradeAmount, extraData.length != 0);
+
+    _verifyRfqExecute(lowerStrike, higherStrike, maxFee);
+  }
+
+  /// @dev the following function validates whether the maker is selling or buying a spread
+  /// The following table determines whether or not the high strike should be sold or bought
+  ///
+  /// isCallSpread | isLongSpread | isTaker | makerShouldBeSellingSpread
+  ///      0       |       0      |    0    |   0 (higherStrike should be < 0)
+  ///      0       |       0      |    1    |   1 (higherStrike should be > 0)
+  ///      0       |       1      |    0    |   1
+  ///      0       |       1      |    1    |   0
+  ///      1       |       0      |    0    |   1
+  ///      1       |       0      |    1    |   0
+  ///      1       |       1      |    0    |   0
+  ///      1       |       1      |    1    |   1
+  function _verifyHigherStrikeAmount(int strikeAmount, bool isTaker) internal view {
+    PPTSAParams memory params = _getPPTSAStorage().ppParams;
+    bool makerShouldBeSellingSpread = (params.isCallSpread == params.isLongSpread) ? isTaker : !isTaker;
+    if (makerShouldBeSellingSpread && strikeAmount <= 0) {
+      revert PPT_InvalidHighStrikeAmount();
+    } else if (!makerShouldBeSellingSpread && strikeAmount >= 0) {
+      revert PPT_InvalidHighStrikeAmount();
+    }
+  }
+
+  function _createStrikes(IRfqModule.TradeData[] memory makerTrades)
+    internal
+    view
+    returns (StrikeData memory lowerStrike, StrikeData memory higherStrike)
+  {
+    // TODO: Add check that expiries are the same at that the expiries in here are above min expiry and below max expiry
+    if (makerTrades.length != 2 || makerTrades[0].asset != makerTrades[1].asset) {
+      revert PPT_InvalidParams();
+    }
+    StrikeData memory strike1 = _createStrikeData(makerTrades[0]);
+    StrikeData memory strike2 = _createStrikeData(makerTrades[1]);
+    if (strike1.strike > strike2.strike) {
+      return (strike2, strike1);
+    }
+    return (strike1, strike2);
```

Comment:
This is checked in the strike diff logic. Strike diff is required to be a number above 2e20 right now so this would cover when both strikes are equal.